server/images.go

```diff
@@ -244,9 +240,24 @@ func GetModel(name string) (*Model, error) {
 		}
 	}
 
+	runnerDigest, err := runnerDigest(model)
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculate runner digest: %v", err)
+	}
+	model.RunnerDigest = runnerDigest
+
 	return model, nil
 }
 
+func runnerDigest(m *Model) (string, error) {
+	encodedModel, err := json.Marshal(m)
```

Comment:
Is `json` required here or can we just hash the relevant fields string-concatenated? I don't think there's a collision

Reply:
The json is cleaner here imo because the fields of the model may be more complex than just strings (eg: maps, and slices of interfaces). Concatenating those would be more verbose:
```
func concatenateFields(m Model) string {
	var parts []string

	fields := []stringm.OriginalModel, m.System}
	parts = append(parts, fields...)
	parts = append(parts, m.AdapterPaths...)

	var optionKeys []string
	for k := range m.Options {
		optionKeys = append(optionKeys, k)
	}
	sort.Strings(optionKeys)

	for _, key := range optionKeys {
		valStr, _ := json.Marshal(m.Options[key])
		keyValConcat := key + ":" + string(valStr)
		parts = append(parts, keyValConcat)
	}

	for _, emb := range m.Embeddings {
		parts = append(parts, emb.String())
	}

	return strings.Join(parts, "")
}
```